client/client.go

```diff
@@ -45,12 +45,21 @@ func (h Client) RunPipeline(request model.PostPipelineRequest) (pipeline *model.
 		if pipeline.State.Result.OK() {
 			break
 		}
+
 		time.Sleep(time.Second * 5) // TODO
 		pipeline, err = h.GetPipeline(model.GetPipelineRequest{
 			Workspace:  request.Workspace,
 			Repository: request.Repository,
 			Pipeline:   pipeline,
 		})
+
+		if err != nil {
+			return nil, err
+		}
+
+		if pipeline.State.Result.HasError() {
+			panic(fmt.Sprintf("Pipeline finished with error %v", pipeline.State.Result.Error))
```

Comment:
I don't think we should be including a panic within the lib itself, we should allow the app using it to decide when to panic, this could have adverse effects on the Terraform provider we've written and the BPR CLI.

I think returning a new error would be better.

`return nil, fmt.Errorf("Pipeline finished with error %v", pipeline.State.Result.Error)`